// ng-recaptcha/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var recaptcha_component_1 = require("./recaptcha/recaptcha.component");
exports.RecaptchaComponent = recaptcha_component_1.RecaptchaComponent;
var recaptcha_loader_service_1 = require("./recaptcha/recaptcha-loader.service");
exports.RecaptchaLoaderService = recaptcha_loader_service_1.RecaptchaLoaderService;
exports.RECAPTCHA_LANGUAGE = recaptcha_loader_service_1.RECAPTCHA_LANGUAGE;
exports.RECAPTCHA_BASE_URL = recaptcha_loader_service_1.RECAPTCHA_BASE_URL;
exports.RECAPTCHA_NONCE = recaptcha_loader_service_1.RECAPTCHA_NONCE;
var recaptcha_module_1 = require("./recaptcha/recaptcha.module");
exports.RecaptchaModule = recaptcha_module_1.RecaptchaModule;
var recaptcha_settings_1 = require("./recaptcha/recaptcha-settings");
exports.RECAPTCHA_SETTINGS = recaptcha_settings_1.RECAPTCHA_SETTINGS;
var recaptcha_v3_module_1 = require("./recaptcha/recaptcha-v3.module");
exports.RecaptchaV3Module = recaptcha_v3_module_1.RecaptchaV3Module;
var recaptcha_v3_service_1 = require("./recaptcha/recaptcha-v3.service");
exports.ReCaptchaV3Service = recaptcha_v3_service_1.ReCaptchaV3Service;
exports.RECAPTCHA_V3_SITE_KEY = recaptcha_v3_service_1.RECAPTCHA_V3_SITE_KEY;
